R/gg-plots.R

```diff
@@ -1292,6 +1292,108 @@ ggally_ratio <- function(
 
 
 
+#' Plots the number of observations
+#'
+#' Plot the number of observations by using rectangles
+#' with proportional areas.
+#'
+#' @param data data set using
+#' @param mapping aesthetics being used
+#' @param ... other arguments passed to \code{\link[ggplot2]{geom_tile}(...)}
+#' @author Joseph Larmarange \email{joseph@@larmarange.net}
+#' @keywords hplot
+#' @export
+#' @examples
+#' data(tips, package = "reshape")
+#' ggally_count(tips, mapping = ggplot2::aes(x = smoker, y = sex))
+#'
+#' ggally_count(
+#'   as.data.frame(Titanic),
+#'   mapping = ggplot2::aes(x = Class, y = Survived, weight = Freq)
+#' )
+ggally_count <- function(data, mapping, ...) {
+
+  mapping <- mapping_color_to_fill(mapping)
+  vars <- unlist(lapply(mapping, mapping_string))
+  x_var <- vars["x"]
+  y_var <- vars["y"]
+  if (is.na(x_var)) stop("'x' aesthetic is required.")
+  if (is.na(y_var)) stop("'y' aesthetic is required.")
+
+  if (is.na(vars["weight"])) {
+    data$.weight <- 1
+  } else {
+    data$.weight <- data[[vars["weight"]]]
+  }
+
+  # compute formula for aggregation
+  # exclude weight and do not double count a var mapped to several aesthetics
+  f <- stats::as.formula(paste(
+    ".weight ~",
+    paste(unique(vars[names(vars) != "weight"]), collapse = "+")
+  ))
+
+  d <- stats::aggregate(f, data, sum, na.rm = TRUE)
+  names(d)[which(names(d) == ".weight")] <- ".n"
+
+  d$.x <- d[[x_var]] # keep original x &nd y
+  d$.y <- d[[y_var]]
+  d[[x_var]] <- as.numeric(d$.x)
```

Comment:
We need to be careful to still work when the columns don't exist in the data.

```r
ggally_count(tips, mapping = ggplot2::aes(x = toupper(smoker), y = toupper(sex)))
```

Might need to use a hard coded location like how you did `data$.x_copy` below.

Reply:
will explore and check

Reply:
fix by https://github.com/ggobi/ggally/pull/321/commits/b9c71817857e713ffc4a8edf0de8ed2f35201860